dashboard/app/models/user.rb

```diff
@@ -285,6 +285,8 @@ class User < ApplicationRecord
     Services::Lti.create_lti_user_identity(self)
   end
 
+  after_create :verify_teacher!, if: -> {teacher? && Policies::Lti.lti?(self)}
```

Comment:
should this be `before_save` for the times when a teacher's Code.org LTI account was accidentally created as a student account and later updated to a teacher account when we do another roster sync?

If it should be done only on create, should this be `before_create` so the changes are automatically saved? Right now I'm suspicious this doesn't actually save the changes to the DB because the modification is happening after the create has happened.

Another thought, does this also need to handle roster syncs where a teacher becomes a student? I can't remember if that is something we support. We should probably unverify the account if they become a student.

Reply:
The user must be persistent to be granted permission:
https://github.com/code-dot-org/code-dot-org/blob/9cee0fc395242a7c0f26363bc61eaa3e31cac80e/dashboard/app/models/concerns/user_permission_grantee.rb#L28-L32